### ISSUE TITLE
Not all mounts are used

### ISSUE THREAD

**Issue by rngtng** [gopasspw/gopassbridge#131]
Hi

I've a gopass setup working with _private_ (root) and _work_ store like this:

```
> gopass mounts
gopass (/Users/foobar/.password-store)
└── work (/Users/foobar/.password-store-work)
```

Still, _gopassbridge_ wont show any secrets from _work_ store, only root.  Whereas `gopass show` returns me secret from both stores, what did I wrong?

Cheers

**Comment by martinhoefling** [comment]
Several people including me use it with multiple stores. Can you give some more details, e.g. gopass version / layout of the store? You're searching in the text field right?

**Comment by rngtng** [comment]
@martinhoefling thx for quick reply. yes it's search via text field. I noticed gopass UI has same problem, so maybe it's my gopass version (1.8.6) which doesn't expose my stores correctly via jsonapi?

Layout is like given above.

**Comment by martinhoefling** [comment]
Gopass UI isn't using the API afaik. Can you give me some more details on the contents of the store or create a minimal example? On what platform? I have no issues with 1.8.6 sofar.

**Comment by rngtng** [comment]
hm weird, so maybe a gopass issue with exposing mounts in general? 
I'm running on latest macosx, both stores contain std website logins. `gopass show` offers me to encrypt any keys from both stores.. I first setup a local store and then added a second mount. Maybe worth to share: both mount have the same remote, but operate on a different branch...

**Comment by martinhoefling** [comment]
Can you please use https://github.com/gopasspw/gopassbridge/tree/master/test-client and query for an item, that you find and one that you don't find? And please also attach the output of `gopass version`.

**Comment by rngtng** [comment]
hm strange with this test client it all looks good, keys from both stores are returned. even when passing a query both stores are searched.. same query on chrome extension is empty..

```
gopass version
gopass 1.8.6 (d5b0d3b906cdd9f16ad3f21e366845af7f2c22f3 2019-10-11 18:54:23) go1.13.1 darwin amd64
<root>     - gpg 2.2.18 - git 2.24.0 -   fs 0.1.0
Available Crypto Backends: gpgcli, openpgp, plain, xc
Available RCS Backends: gitcli, noop
Available Storage Backends: fs, inmem
```

**Comment by martinhoefling** [comment]
Which browsers did you try? What OS / version? Can you try to run it in dev set up to debug?

**Comment by rngtng** [comment]
@martinhoefling latest chrome (Version 79.0.3945.79), latest MacOX (10.15.2) can you quick assist me on how to run **gopass** and/or **gopassbridge** in dev setup for debugging, pls?

**Comment by martinhoefling** [comment]
You can try the instructions [here](https://github.com/gopasspw/gopassbridge#development) as a starter. More information on how to debug extensions can be found [here](https://extensionworkshop.com/documentation/develop/debugging/)

**Comment by rngtng** [comment]
@martinhoefling Thanks, I followed your steps. Good news, I found the issue:

TL;DR: chrome doesn't set `XDG_CONFIG_HOME` env var, which in my case is set to `~/dotfile`. As of this my `gopass.yml` is not sourced, and therefore no additional mounts loaded.

First, by following your development steps I  tested the extension in firefox `make run-firefox' - again it all worked. Loaded the extension in devmode  in chrome - still no luck, only main mount exposed. In next step updated `gopass_wrapper.sh` to intercept in/outputs and activate gopass debug outputs:

```
tee in.log | GOPASS_DEBUG=true /usr/local/bin/gopass jsonapi listen | tee out.log
```

And finally pointed the `query-jsonapi.sh` testscript to wrapper too:

```
echo '{"type":"query", "query": "s"}' | ./test-client | ~/dotfiles/gopass/gopass_wrapper.sh
```

Now I re-ran the tests in chrome, firefox & test script and compared the output. It clearly showed in FF & testscript, the correct `gopass.yml` file is sourced. Only in chrome it's not:

```
[DEBUG] Trying to load config from ~/.config/gopass/config.yml
[DEBUG] Trying to load config from ~/.gopass.yml
```
Bot files don't exists, so gopass must have loaded just my default mount `~/.password-store` which explains why the second mount is not returned. Looking into `env` I found the reason:
in FF and testscript, my usual env is loaded with `XDG_CONFIG_HOME` set to `~/dotfiles`, although in chrome case, only very few env variables are set, with `XDG_CONFIG_HOME` being empty. This explains why my right gopass config isn't sourced. Guess related is https://github.com/gopasspw/gopass/issues/46

So, how do I get chrome to set `XDG_CONFIG_HOME` ? :)

**Comment by martinhoefling** [comment]
Glad that you were able to debug this. One option would be to set it in the file (at install time) but that is prone to errors when changed later. I wonder if we could aid in debugging those issues somehow.

**Comment by rngtng** [comment]
quick update: FF did only work as I started it via `make run-firefox`. If I open FF the usual way it shows same behaviour as chrome. Apparently applications not started via CLI have a different environment/context. 'causing `XDG_CONFIG_HOME` to be empty. So not a chrome issue, more a general issue of my setup. As of this I solved it by creating a symlink to my special path:

```
ln -s ~/dotfiles/gopass/config.yml .gopass.yml
```

For debugging an earlier feedback on which config is sourced would have helped. But dunno how this could have been exposed via jsonapi. 

Thx for your support closing this issue, as IMHO nothing to fix on gopassbridge or gopass side...

**Comment by varac** [comment]
I experienced the same, gopass-bridge didn't show any password from my mounted stores. I noticed that I only had the config file `~/.config/gopass/config`, which is used by gopass cli.
When I also added `~/.config/gopass/config.yml` gopass-bridge showed my mounts as it should.
So I'm utterly confused by these 2 config files. The [gopass config doc](https://github.com/gopasspw/gopass/blob/master/docs/config.md#configuration-options) only talks about the `config`, not the `config.yml`. Should gopass-bridge also read `config` instead of `config.yml` ?
